python/src/pyarrow/common.cc

```diff
@@ -45,18 +45,26 @@ MemoryPool* get_memory_pool() {
 }
 
 // ----------------------------------------------------------------------
-// PyBytesBuffer
+// PyBuffer
 
-PyBytesBuffer::PyBytesBuffer(PyObject* obj)
+PyBuffer::PyBuffer(PyObject* obj)
     : Buffer(reinterpret_cast<const uint8_t*>(PyBytes_AS_STRING(obj)),
-          PyBytes_GET_SIZE(obj)),
+      PyBytes_GET_SIZE(obj)),
```

Comment:
These PyBytes_* APIs need to be removed. If you write a test case for a non-bytes object exporting the buffer interface then this will fail